src/data/markdown/docs/20 jslib/01 jslib/01 aws/EventBridgeClient/putEvents.md

```diff
@@ -0,0 +1,70 @@
+---
+title: 'EventBridgeClient.putEvents'
+description: 'EventBridgeClient.putEvents sends custom events to Amazon EventBridge'
+excerpt: 'EventBridgeClient.putEvents sends custom events to Amazon EventBridge'
+---
+
+`EventBridgeClient.putEvents` sends custom events to Amazon EventBridge so that they can be matched to rules.
+
+### Parameters
+
+| Parameter     | Type            | Description                                                                                                              |
+| :------------ | :-------------- | :----------------------------------------------------------------------------------------------------------------------- |
+| input         | [PutEventsInput](#puteventsinput) | An array of objects representing events to be submitted. |
```

Comment:
I'm looking at both `input` and `Entries` with the same description, and I'm wondering if this would be more accurate if it said: "An object with the key `Entries` and the value of an array of EventBridgeEntry, representing events to be submitted.", or something along these lines.

Wdyt? Or is this more confusing?

Reply:
I see what you mean, but the `PutEventsInput` is a data interface defined by AWS, which we also define in the code, and which also has other attributes such as `EndpointId` (and I believe more that we haven't included yet).

Thus, I believe it is better kept this way.

However, I'm not happy with how we deal with these nested type definitions in the docs at the moment. The problem is that if the argument to a function is an object with a strict form and shape that we expect, it's hard to describe it to the user in a concise way. Having multiple tables, as I did here, was my best bet. We could also have dedicated separate pages, but that makes documenting a library like AWS very cumbersome.

What do you think? Maybe you see an alternative that would be more friendly to both the user and the documentation writer? 😄 🙇🏻

Reply:
That's a great point... And I think tables are an ok solution for right now, feel free to merge this PR.

I'll open a separate issue so we can discuss/test some different ways of how to better display nested objects, maybe we can look at some other docs and get some ideas. :)